pkg/cli/image/extract/extract.go

```diff
@@ -593,22 +596,32 @@ func (n *copyFromDirectory) Alter(hdr *tar.Header) (bool, error) {
 type copyFromPattern struct {
 	Base string
 	Name string
+	RootDir bool
 }
 
-func newCopyFromPattern(dir, name string) archive.AlterHeader {
+func newCopyFromPattern(dir, name string, rootDir bool) archive.AlterHeader {
+	if rootDir {
+		return &copyFromPattern{Name: name, RootDir: true}
+	}
 	if !strings.HasSuffix(dir, "/") {
 		dir = dir + "/"
 	}
-	return &copyFromPattern{Base: dir, Name: name}
+	return &copyFromPattern{Base: dir, Name: name, RootDir: false}
 }
 
 func (n *copyFromPattern) Alter(hdr *tar.Header) (bool, error) {
-	if !changeTarEntryParent(hdr, n.Base) {
-		return false, nil
-	}
 	matchName := hdr.Name
-	if i := strings.Index(matchName, "/"); i != -1 {
-		matchName = matchName[:i]
+	if n.RootDir {
+		if !changeTarEntryName(hdr, n.Name) {
+			return false, nil
+		}
```

Comment:
add line  `matchName = hdr.Name` here

Reply:
Ok now I see those reviews are connected as one :) sorry. I am just updating it and will send new review request :P Thanks for your patience!!!